kolibri/core/auth/serializers.py

```diff
@@ -137,6 +139,25 @@ class Meta:
         fields = ("id", "name")
 
 
+class CreateFacilitySerializer(serializers.ModelSerializer):
+    preset = serializers.ChoiceField(choices=facility_presets.choices)
+
+    class Meta:
+        model = Facility
+        fields = ("id", "name", "preset")
+
+    def create(self, validated_data):
+        with transaction.atomic():
+            facility_dataset = FacilityDataset.objects.create(
+                preset=validated_data.get("preset")
+            )
+            facility = Facility.objects.create(
+                name=validated_data.get("name"), dataset=facility_dataset
+            )
+            facility.dataset.reset_to_default_settings(validated_data.get("preset"))
+        return facility
+
```

Comment:
@Jaspreet-singh-1032  thanks for adding this. However, here are a few more comments to further make the code more robust;

- You could set the `preset` and `name` fields as variables instead eg.
  - `preset = validated_data.get("preset")`
- Then add some validation checks to these variables before proceeding to use them. You can alternatively use the `validate` method within the serializer class if you like
- It's also a good idea to wrap the code around a try..catch block just in case any of the transactions fail

Reply:
Hi @akolson, I've made the changes you suggested by setting `preset` and `name` fields as variables. However, I have not added additional validation to these variables because I believe the data is already being validated at this point. if there are specific scenarios or edge cases where you think further validation is needed, please let me know, and I'll be happy to implement it accordingly. And I have also added the try-catch block.

Reply:
I see the `name` is being validated on the frontend, it's probably good to add this validation on the backend too. But, I could be lacking a bit of context overall so I am happy to proceed with the merge(Changes look good, thanks :)) as long as @rtibbles  is happy with the changes.

Reply:
By default the serializer will do introspection of the Django model, so that should be doing some validation to ensure it doesn't exceed the maximum length of the name attribute, and that it is a string, so I think Django REST Framework magic is probably doing the leg work for us here!